doc/source/dev/build_a_job_runner.rst

```diff
@@ -350,7 +350,7 @@ The following is a generic code snippet for the ``recover`` method.
 .. code-block:: python
 
     ajs = AsynchronousJobState(files_dir=job_wrapper.working_directory, job_wrapper=job_wrapper)
-    ajs.job_id = str(job_wrapper.job_id)
+    ajs.job_id = str(job_wrapper.get_job().job_runner_external_id)
```

Comment:
```suggestion
    asj.job_id = job.get_job_runner_external_id()
```

You're right! However, since we already have access to the `Job` object (it's in the `recover` method signature), we might as well use it; that will also keep it consistent with other runners' implementation. Thanks for catching this! (I'm afraid these docs may be outdated...)

Reply:
A related question: is there a difference between ```job.id``` and ```job_wrapper.id``` ? Both are used in the recover method in several examples.

Reply:
There shouldn't be. A JobWrapper takes a `model.Job` object as an argument at construction time - which is where its `job_id` is assigned. Can you point me to an example when they are used in the same code block?

Reply:
Here for instance in the recover method https://github.com/galaxyproject/galaxy/blob/7e1971e89f75699f30ad7de8a135529b8c3c5c5c/lib/galaxy/jobs/runners/godocker.py#L250

Se example of using L255 and L265 - but I guess I heard that the godocker is outdated. Is it only that this is the example typically used in the docs.

Reply:
I don't know. Furthermore, I don't know why we are using `job_id = job_wrapper.job_id` instead of the external job id like we do in other runners. That runner was added in #2653; although it's not ancient, my guess that's a bug. But I may be wrong. But I may be wrong.. @mvdbeek am I missing something here?